### PR TITLE
osac-project: add edge04 unified vmaas test job

### PR DESCRIPTION
**Pull request by obochan-rh**
# osac-project: Add edge04 unified vmaas test job

This PR updates the OpenShift CI configuration (openshift/release → osac-project) to add a new PR presubmit that runs the full vMAAS test suite against the persistent [edge-04](https://redhat.atlassian.net/browse/edge-04) baremetal host.

What changed, in practical terms
- Added a new presubmit test: e2e-edge04-vmaas-all-pr in ci-operator/config/osac-project/osac-test-infra/osac-project-osac-test-infra-main.yaml. The job requires the intranet capability and uses the new osac-project-edge04 workflow.
- Added a new workflow (ci-operator/step-registry/osac-project/edge04/osac-project-edge04-workflow.yaml) that sequences a wait step then the osac-project-edge04-test step and runs osac-project-gather as a post step. (The PR replaces an inline sleep with a wait step to improve debug access timing.)
- Added a test step ref (ci-operator/step-registry/osac-project/edge04/test/osac-project-edge04-test-ref.yaml) that:
  - Runs a dev-scripts-based command with a 3h timeout and 10m grace period.
  - Mounts credentials (edge04key) at /tmp/secrets/edge04.
  - Exposes env vars for EDGE04_HOST/EDGE04_USER and vMAAS test configuration (namespace, VM/cluster templates).
- Added the remote execution script (ci-operator/step-registry/osac-project/edge04/test/osac-project-edge04-test-commands.sh) which SSHes to the configured [edge-04](https://redhat.atlassian.net/browse/edge-04) host, validates kubeconfig, then runs the specified OSAC test image via podman (passing kubeconfig, pull secret, workdir) to execute make test-vmaas and copy test reports back.
- Ownership metadata: added OWNERS and metadata.json files to set osac-cicd as approvers/reviewers for the new workflow, step registry, and test ref.

Why this matters
- The new job validates vMAAS behavior directly against an existing persistent [edge-04](https://redhat.atlassian.net/browse/edge-04) baremetal host ([edge-04](https://redhat.atlassian.net/browse/edge-04).edge.lab.eng.rdu2.redhat.com) over SSH, without provisioning a temporary cluster. This provides faster end-to-end validation of the full vMAAS suite for PRs that affect the osac-project test infra.
- Clear ownership (osac-cicd) and a wait step for debug improve maintainability and interactive debugging during presubmit runs.

Files added/updated (high-level)
- ci-operator/config/osac-project/osac-test-infra/osac-project-osac-test-infra-main.yaml — add e2e-edge04-vmaas-all-pr
- ci-operator/step-registry/osac-project/edge04/ — added workflow YAML + metadata, test ref YAML + metadata, test commands script, and OWNERS files

Behavioral notes
- Job runs sequential tests inside a single persistent environment; it uses SSH keys from the test-credentials secret and expects kubeconfig & pull secret to be available on the remote host.

### CODE REVIEW

File: ci-operator/step-registry/osac-project/edge04/osac-project-edge04-workflow.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+workflow:
+  as: osac-project-edge04
+  steps:
+    allow_best_effort_post_steps: true
+    allow_skip_on_success: true
+    test:
+      - ref: wait
+      - ref: osac-project-edge04-test
+    post:
+      - ref: osac-project-gather
```
Comment (shaior) [inline ci-operator/step-registry/osac-project/edge04/osac-project-edge04-workflow.yaml:10]: wait step should be in post, before or after the gather
